blocks/library/embed/index.js

```diff
@@ -21,57 +23,141 @@ registerBlock( 'core/embed', {
 	category: 'common',
 
 	attributes: {
-		url: attr( 'iframe', 'src' ),
 		title: attr( 'iframe', 'title' ),
 		caption: children( 'figcaption' ),
 	},
 
-	edit( { attributes, setAttributes, focus, setFocus } ) {
-		const { url, title, caption } = attributes;
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.doServerSideRender = this.doServerSideRender.bind( this );
+			// Copies the block's url so we can edit it without having the block update
+			this.state = {
+				oEmbedHtml: '',
+				error: false,
+				fetching: false,
+			};
+			this.noPreview = [
+				'facebook.com',
+			];
+			this.videoEmbeds = [
+				'youtube.com',
+				'youtu.be',
+				'vimeo.com',
+				'dailymotion.com',
+				'dai.ly',
+				'hulu.com',
+				'wordpress.tv',
+				'funnyordie.com',
+				'collegehumor.com',
+			];
+		}
 
-		if ( ! url ) {
-			return (
-				<Placeholder icon="cloud" label={ wp.i18n.__( 'Embed URL' ) } className="blocks-embed">
-					<input type="url" className="placeholder__input" placeholder={ wp.i18n.__( 'Enter URL to embed here...' ) } />
-					<Button isLarge>
-						{ wp.i18n.__( 'Embed' ) }
-					</Button>
-				</Placeholder>
+		doServerSideRender( event ) {
+			event.preventDefault();
+			const api_url = '/?rest_route=/gutenburg/v1/render/';
+			const { url } = this.props.attributes;
+			const form = new FormData();
+			// core/oembed is a server side block for rendering the WP_oEmbed functionality
+			const oembed_block = serializeBlock( {
+				type: 'core/oembed',
+				attributes: { url },
+				settings: {},
+				saveContent: '',
+			} );
+			form.append( 'content', oembed_block );
+			this.setState( { error: false, fetching: true } );
+			fetch( api_url, {
```

Comment:
For me it fails to embed because this is not the proper path. Needs path to the WP site. https://developer.wordpress.org/reference/functions/rest_url/